lib/topic_view.rb

```diff
@@ -512,8 +518,8 @@ def links
     @links ||= TopicLink.topic_map(@guardian, @topic.id)
   end
 
-  def user_post_bookmarks
-    @user_post_bookmarks ||= @topic.bookmarks.where(user: @user)
+  def user_bookmarks
+    @user_bookmarks ||= Bookmark.for_user_in_topic(@user&.id, @topic.id)
```

Comment:
I don't think we need to execute a query when for an anon user. We can just return an empty array early if `@user&.id` is `nil`

Reply:
Just curious, what is the difference between `user_bookmarks` and `bookmarks`?

Reply:
Oh yeah you're right...will remove one. And as for the query for anon, I will return an empty array, might just need some extra checks higher up the chain or some optional param for ordering, because I think in one place we order this

Reply:
Never mind about the query thing, I was reading the wrong method! Did the rename though, and we don't have to worry about anon because the `has_bookmarks?` method protects us there :+1: